src/algebra/order/hom/monoid.lean

```diff
@@ -171,6 +171,20 @@ lemma map_nonpos (ha : a ≤ 0) : f a ≤ 0 := by { rw ←map_zero f, exact orde
 
 end ordered_add_comm_monoid
 
+section ordered_add_comm_group
+variables [ordered_add_comm_group α] [ordered_add_comm_monoid β] [add_monoid_hom_class F α β]
+  {f : F}
+
+lemma monotone_iff_map_nonneg : monotone (f : α → β) ↔ ∀ a, 0 ≤ a → 0 ≤ f a :=
```

Comment:
Are the following true?
```
lemma strict_mono_iff_map_pos : strict_mono (f : α → β) ↔ ∀ a, 0 < a → 0 < f a := sorry
lemma antitone_iff_map_nonpos : antitone (f : α → β) ↔ ∀ a, 0 ≤ a → f a ≤ 0 := sorry
lemma strict_anti_iff_map_neg : strict_anti (f : α → β) ↔ ∀ a, 0 < a → f a < 0 := sorry
```

Reply:
Okay I've added these lemmas and their dual versions, for a total of 8 lemmas. The strict lemmas require strict monotonicity of addition, which is satisfied by `ordered_cancel_add_comm_monoid` but nothing weaker AFAIK (or should I use `covariant_class β β (+) (<)`)?

I followed your naming convention so that (non)pos/neg always refers to `f a` and not to `a`, but maybe other reviewers will have different opinions.

Reply:
You probably should use `covariant_class β β (+) (<)`, yes.